sdk/typescript/src/signers/txn-data-serializers/call-arg-serializer.ts

```diff
@@ -28,6 +28,14 @@ export class CallArgSerializer {
   async extractObjectIds(txn: MoveCallTransaction): Promise<ObjectId[]> {
     const args = await this.serializeMoveCallArguments(txn);
     return args
+      .map((arg) =>
+        'ObjVec' in arg
+          ? Array.from(arg.ObjVec).map((a) => ({
+              Object: a,
+            }))
+          : arg
+      )
+      .flat()
```

Comment:
why flat() here?
I am thinking if the original args are sth like
[obj_id1, [obj_id2, obj_id3], obj_id4], with the flat(), the serialization result is the same as that of [obj_id1, obj_id2, obj_id3, obj_id4]?

Reply:
Here the goal is to extract input object ids, not for serialization.